engine/buff/buff.cpp

```diff
@@ -829,6 +831,76 @@ buff_t* buff_t::set_duration_multiplier( double multiplier )
   return this;
 }
 
+buff_t* buff_t::set_time_modifier( double modifier )
```

Comment:
While it's not needed for this Rogue case, but you should add a comment here or somewhere else to let people know that this does not apply the modifications to buff tick times and existing tick events that would appear in game with modify time rate effects.

Reply:
I think this is a fairly important point that I didn't catch. If this code was in the rogue module I don't think the lack of tick time modification is really an issue, but in simc core it should work correctly with everything buff_t supports.

Reply:
I was looking into doing tick time modification, but had two (three) issues:

0. I am horribly unfamiliar with all this, and absolutely need direction to take this much farther.

1. It's basically impossible to test weather the behavior matches game behavior, since the modifier isn't applied to anything that ticks right now (slice and dice's healing was moved into a separate buff to head off this issue (see spell 426605 which now holds the heal separately from the main buff and has no duration of its own, plus a fixed 2s period), adrenaline rush just modifies stats, blade flurry just procs on damage).

2. If we assume the time modifier speeds up/slows down ticks like it does the buff's expiry, we need to calculate tick times using the "apparent" buff duration and not the real-time buff duration (and then scale those into the real-time domain using the time modifier), and dealing with real-time vs apparent buff durations _at all_ seems like a massive hassle for something that doesn't seem to matter just yet. Minimally, I have no idea what other parts of the sim should be looking at real-time vs apparent-time durations. 
    * Not dealing with it technically means slice and dice's pandemic refresh when paused is bugged in the PTR rogue sim I've been working on (since it should probably pandemic at 1200 real-seconds left while paused and not 12s - this is the in-game behavior, the pandemic duration is definitely based on the apparent duration), but that'll never happen in any real sim, since the sim'll never intentionally cast slice with anything close that much real-time left (why would you, massive waste of resources).

Reply:
> It's basically impossible to test weather the behavior matches game behavior, since the modifier isn't applied to anything that ticks right now

I've since learned the old https://www.wowhead.com/spell=320919/echoes-of-elisande anima power actually uses this modifier (as a positive number). I vaguely remember playing a mage in torghast with it, so I do think it multiplies the tickrate of all hots and dots. And cooldowns. And other stuff, for the full applies-to-everything effect. And, iirc, weird stuff like environmental damage ticks, too.

I still don't have a good grasp of everything that'd need to change to properly support it, though, especially what needs the real-time vs apparent-time distinction.